src/routing_table/mod.rs

```diff
@@ -259,8 +260,8 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
-    pub fn our_name(&self) -> &T {
-        &self.our_name
+    pub fn our_name(&self) -> T {
+        self.our_name
```

Comment:
But `name` is a `u32`, `u256` or similar? Surely it's cheaper to copy than reference?